_posts/2017/2017-12-11-set-title-in-the-view-template-using-content-for.md

```diff
@@ -0,0 +1,71 @@
+---
+title: 'Set your page title in the view template using content_for'
+description: 'Cut out all that instance variable nonsense.'
+layout: article
+category: ruby
+image:
+  base: '2017/set-title-in-the-view-template-using-content-for'
+  alt: 'Books'
+  credit: 'Annie Spratt'
+  source: 'https://unsplash.com/photos/-9vMBjrU-RA'
+---
+
+Rails has `content_for` as it’s primary way to store content in one place for use in other views, layouts or helpers. See `content_for` in the Rails documentation at [in the ActionView helpers](http://api.rubyonrails.org/v5.1.3/classes/ActionView/Helpers/CaptureHelper.html#method-i-content_for).
+
+## Instead of…
+
+...using an instance variable in the controller.
+
+### `things_controller.rb`
+
+```ruby
+def show
+  @page_title = 'Page Title'
+  # other controller stuff
+end
+```
+
+### And `views/layouts/application.html.erb`
+
+In the `<head>`.
+
+```erb
+<title><%= @page_title || 'Default' %></title>
+```
+
+
+## Use…
+
+...`content_for` and `yield`.
+
+### `things/show.html.erb`
+
+I like to put the title at the top of the individual view file.
+
+```erb
+<% content_for(:html_title) { 'Title' } %>
+```
+
+### And `views/layouts/application.html.erb`
+
+In the `<head>`.
+
+```erb
+<title><%= content_for?(:html_title) ? yield(:page_title) : "Default" %></title>
+```
+
+
+## But why?
+
+Using `content_for` in this way keeps information about how to render the page inside the relevant view template. It also avoids cluttering the controller with non-business logic.
+
+You also avoid creating yet another 'magic' instance variable that gets automatically revealed to the view by Rails.
+
+Another heuristic for using this technique is that when rendering JSON and/or AJAX from the same restful action, those versions don't need a page title, indicating it belongs in the HTML view template.
```

Comment:
*RESTful? (I dunno what the convention is..maybe you're right!)

Reply:
I think what you're trying to say is:

> Here's another way you know I'm right about how you should set your page titles. Rending JSON or AJAX doesn't require adding a page title. Therefore, it's clear that the info should live with the relevant view.

Is that correct?

If so, I think you can reword this paragraph to be more straightforward.